lib/IDE/CodeCompletionResultBuilder.h

```diff
@@ -159,8 +130,23 @@ class CodeCompletionResultBuilder {
     Flair |= Options;
   }
 
-  void setExpectedTypeRelation(CodeCompletionResultTypeRelation relation) {
-    ExpectedTypeRelation = relation;
+  /// Indicate that the code completion item does not produce something with a
+  /// sensible result type, like a keyword or a method override suggestion.
+  void setResultTypeNotApplicable() {
+    ResultType = CodeCompletionResultType::notApplicable();
+  }
+
+  /// Set the result type of this code completion item and the context that the
+  /// item may be used in.
+  /// This is not a single unique type because for code completion we consider
+  /// e.g. \c Int as producing both an \c Int metatype and an \c Int instance
+  /// type.
+  void setResultTypes(ArrayRef<Type> ResultTypes,
+                      const ExpectedTypeContext &TypeContext,
+                      const DeclContext *DC) {
+    this->ResultType = CodeCompletionResultType(ResultTypes);
+    this->TypeContext = TypeContext;
+    this->DC = DC;
```

Comment:
Result type is a "context free" data whereas `TypeContext` and `DC` are contextual. Setting these in a single method seems weird.

Reply:
I agree. I split the method into two: `setResultType` and `setTypeContext`